### ISSUE TITLE
Proposal: Checklist for adding a plugin

### ISSUE THREAD

**Issue by alitoufighi** [fluent/fluent-bit#2237]
Hi.
I see lots of inconsistencies in where the plugins are introduced.
Sometimes a written and merged plugin doesn't have any documentations.
Also there are documented and present plugins that are not listed in neither [source code Readme](https://github.com/fluent/fluent-bit/blob/master/README.md#plugins-inputs-filters-and-outputs) nor [docs summary for each type of plugins](https://docs.fluentbit.io/manual/pipeline/inputs).

I think having a checklist that must be done before a plugin implementation is merged can be good idea, since IMO, a feature without any usage documentation has no value.

Also we can develop a CI script to update the source code Readme and the docs when a plugin is added (even with an empty page, only to show this software is capable of doing some (unknown) things).

To sum up, my concern is two things here:
* The lack of consistency between the two mentioned places
* The lack of documentation for some plugins
